### PR TITLE
Bring in a Rails 5.1 setup

### PR DESCRIPTION
**Pull request by schmijos**
I recently set up a Rails 5.1 project. This is the setup of it.

### CODE REVIEW

File: templates/bin/setup [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
+  # Install JavaScript dependencies if using Yarn
+  # system('bin/yarn')
 
-  run 'Removing old logs and tempfiles' do
-    system 'rm -f log/*'
-    system 'rm -rf tmp/cache'
+  puts "\n== Copying sample files =="
+  unless File.exist?('config/application.yml')
+    system! 'cp config/application.example.yml config/application.yml'
   end
```
Comment (schmijos) [inline templates/bin/setup]: This is custom for figaro. I myself use dotenv and this configuration:

    unless File.exist?('.env.local')
      cp '.env', '.env.local'
    end

File: templates/bin/setup [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
 
-  run 'Creating pre-push hook link', '.git/hooks/pre-push' do
-    system 'ln -s ../../bin/fastcheck .git/hooks/pre-push'
-  end
+  puts "\n== Preparing database =="
+  system! 'bin/rails db:reset'
```
Comment (schmijos) [inline templates/bin/setup]: Here there's originally written `bin/rails db:setup`. `db:reset` drops it before, which makes sense in a local environment. Do you agree?

File: templates/bin/setup [hunk 3]
```diff
@@ -1,37 +1,37 @@
 #!/usr/bin/env ruby
```
Comment (coorasse) [inline templates/bin/setup]: To me it does not make a lot of sense to have this file here. We need to update it at every Rails change and seems a bit stupid. Why don't we remove it?

File: templates/bin/setup (hunk 1 above)
Comment (coorasse) [reply templates/bin/setup]: If the project has figaro, as we suggest in the guide, then this part makes sense. Maybe it should be moved in the figaro part of the guide. this also gives another reason to not keep this entire file here. what do you think?

File: templates/bin/setup (hunk 2 above)
Comment (coorasse) [reply templates/bin/setup]: doesn't matter. the schema drops the tables anyway.

File: templates/bin/setup (hunk 1 above)
Comment (schmijos) [reply templates/bin/setup]: Agreed.

File: templates/bin/setup (hunk 3 above)
Comment (schmijos) [reply templates/bin/setup]: Perfectly fine.

**Pull request merged**